### PR TITLE
Added search by properties to the GenerisSearch

### PR DESCRIPTION
**Pull request by zagovorichev**

### CODE REVIEW

**Comment by krampstudio** [comment]
@jbout could you please have a quick look

**Comment by jbout** [comment]
1. Agreed
2+3: $queryString was originally intended to be search engine dependent and thus a complex user generated string. Building this string programmatically could become problematic, as it would require to define a clear syntax as interface.
If we already have a structured search query then it might be better to extend the interface of the search engine, so that each implementation can treat it 
4. Agreed

**Comment by zagovorichev** [comment]
@krampstudio @jbout Added new method, check it, please. Also was changed ElasticSearch https://github.com/oat-sa/lib-tao-elasticsearch/pull/5

File: models/classes/resources/ListResourceLookup.php [hunk 1]
```diff
@@ -50,8 +50,11 @@ public function getResources(\core_kernel_classes_Class $rootClass, array $selec
     {
         /** @var Search $searchService */
         $searchService = $this->getServiceLocator()->get(Search::SERVICE_ID);
+
         /** @var ResultSet $result */
-        $result = $searchService->query(current($propertyFilters), $rootClass, $offset, $limit);
+        $result = $searchService
+            ->addFiltersByProperties($propertyFilters)
+            ->query(current($propertyFilters), $rootClass, $offset, $limit);
```
Comment (krampstudio) [inline models/classes/resources/ListResourceLookup.php]: Sorry but I still don't get the API. I think I get your intent to not break the existing method `query` but it still odd at the usage.

File: models/classes/search/strategy/GenerisSearch.php [hunk 2]
```diff
@@ -127,4 +132,14 @@ public function supportCustomIndex()
...
         return false;
     }
+
+    /**
+     * (non-PHPdoc)
+     * @see \oat\tao\model\search\Search::addFiltersByProperties()
+     */
+    public function addFiltersByProperties($properties = [])
+    {
+        $this->properties = $properties;
```
Comment (krampstudio) [inline models/classes/search/strategy/GenerisSearch.php]: So it means you set the properties once and all next search using the service instance will also use the same properties.

Comment (zagovorichev) [reply models/classes/search/strategy/GenerisSearch.php]: yep, I've missed this, I could fix it if the idea is generally correct

**Comment by zagovorichev** [comment]
@jbout @krampstudio New implementation is ready, check it, please

File: models/classes/resources/ListResourceLookup.php [hunk 3]
```diff
@@ -78,6 +93,21 @@ public function getResources(\core_kernel_classes_Class $rootClass, array $selec
         ];
     }
 
+    private function getResourceData($resource)
```
Comment (krampstudio) [inline models/classes/resources/ListResourceLookup.php]: Could you please comment this method

File: models/classes/resources/ListResourceLookup.php [hunk 4]
```diff
@@ -48,26 +49,40 @@ class ListResourceLookup extends ConfigurableService implements ResourceLookup
      */
     public function getResources(\core_kernel_classes_Class $rootClass, array $selectedUris = [], array $propertyFilters = [], $offset = 0, $limit = 30)
     {
-        /** @var Search $searchService */
-        $searchService = $this->getServiceLocator()->get(Search::SERVICE_ID);
-        /** @var ResultSet $result */
-        $result = $searchService->query(current($propertyFilters), $rootClass, $offset, $limit);
-        $count = $result->getTotalCount();
+        if (count($propertyFilters) == 1 && isset($propertyFilters[OntologyRdfs::RDFS_LABEL])) {
```
Comment (krampstudio) [inline models/classes/resources/ListResourceLookup.php]: Please add a one line comment explaining that we use the indexed search when only a label is set and the generis search instance when properties are given

**Comment by zagovorichev** [comment]
@krampstudio Update

**Comment by krampstudio** [comment]
This looks ok for me.  I'm waiting @jbout  approval to test and merge.

**Comment by zagovorichev** [comment]
@jbout Update

**Pull request merged**